client/rabbitmq.go

```diff
@@ -0,0 +1,140 @@
+// rabbitmq provides a concrete client implementation using
+// rabbitmq / amqp as a message bus
+
+package client
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	log "github.com/cihub/seelog"
+	"github.com/golang/protobuf/proto"
+	uuid "github.com/nu7hatch/gouuid"
+	"github.com/streadway/amqp"
+	"golang.org/x/net/context"
+
+	"github.com/b2aio/typhon/errors"
+	"github.com/b2aio/typhon/rabbit"
+)
+
+type RabbitClient struct {
+	once       sync.Once
+	inflight   *inflightRegistry
+	replyTo    string
+	connection *rabbit.RabbitConnection
+}
+
+var NewRabbitClient = func() Client {
+	uuidQueue, err := uuid.NewV4()
+	if err != nil {
+		log.Criticalf("[Client] Failed to create UUID for reply queue")
+		os.Exit(1)
+	}
+	return &RabbitClient{
+		inflight:   newInflightRegistry(),
+		connection: rabbit.NewRabbitConnection(),
+		replyTo:    fmt.Sprintf("replyTo-%s", uuidQueue.String()),
+	}
+}
+
+func (c *RabbitClient) Init() {
+	select {
+	case <-c.connection.Init():
+		log.Info("[Client] Connected to RabbitMQ")
+	case <-time.After(10 * time.Second):
```

Comment:
For the next version: let's make this significantly less and/or configurable . In development services that are going to die anyway should die quickly